### PR TITLE
Refactor distributor so that it can used as a torrent-serving library in other applications.

### PR DESCRIPTION
**Pull request by MarkChuCarroll**
- Move distributor.go into new "main" sub-package;
- Move the rest of the code into a package "distributor/torrent".
- Modify API so that users can create trackers, and start and stop
   their goroutines.
- Add a doc.go file to the library package explaining how to use distributor
  as a library.

### CODE REVIEW

File: distributor/distributor.go [hunk 1]
```diff
@@ -32,38 +32,31 @@ func main() {
...
-		logfatal("-serve does not exist: %s", err)
+		torrent.LogFatal("-serve does not exist: %s", err)
 	}
 	if !info.IsDir() {
-		logfatal("-serve is not a directory")
+		torrent.LogFatal("-serve is not a directory")
 	}
 	*dir = filepath.Clean(*dir) // Canonicalize.
-
+	verbosity := torrent.VerbNormal
```
Comment (zorkian) [inline distributor/distributor.go]: But magic constants were awesome! ;) ;)

Kidding. Nice!

File: torrent/logging.go [hunk 2]
```diff
@@ -6,31 +6,37 @@
...
 
-package main
+package torrent
 
 import (
 	"log"
 )
 
-func loginfo(fmt string, args ...interface{}) {
+var VERBOSITY uint32 = 0
```
Comment (zorkian) [inline torrent/logging.go]: So you created the Verbosity type before, but don't use it here? Also, I don't think anything ever calls `SetLoggingVerbosity` below?

Comment (MarkChuCarroll) [inline torrent/logging.go]: Yeah, you're right; I started changing the way I handled verbosity, and then got caught up in something else, and forgot to finish it. Sorry 'bout that.

**Comment by zorkian** [comment]
Thank you for the patch! This is awesome :)

**Pull request merged**